python/feathub/processors/spark/dataframe_builder/source_sink_utils_common.py

```diff
@@ -0,0 +1,38 @@
+#  Copyright 2022 The FeatHub Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from typing import Type
+
+from feathub.feature_tables.feature_table import FeatureTable
+from feathub.feature_tables.sinks.black_hole_sink import BlackHoleSink
+from feathub.feature_tables.sinks.file_system_sink import FileSystemSink
+from feathub.feature_tables.sinks.memory_store_sink import MemoryStoreSink
+from feathub.feature_tables.sinks.print_sink import PrintSink
+from feathub.feature_tables.sources.datagen_source import DataGenSource
+from feathub.feature_tables.sources.file_system_source import FileSystemSource
+
+
+def is_spark_supported_source(source_type: Type[FeatureTable]) -> bool:
```

Comment:
Would it be simpler to add these methods in `source_sink_utils.py`?

Reply:
This design aims to ensure users can check whether they can use a connector by installing Spark BEFORE they actually install Spark. If we move these methods to `source_sink_utils.py`, this file would import `source_sink_utils`, and thus importing Spark dependencies. This would cause import errors even if users only need to use local features.

Reply:
Since we expect this file to only contain methods not depending on pyspark, this means the methods in this file is supposed to be used only by local processor.

Would it be simpler to define these methods in `local_processor.py` similar to `_is_local_file_or_dir`?